server/lib/availabilitySync.ts

```diff
@@ -567,124 +548,158 @@ class AvailabilitySync {
         },
       });
 
+      const filteredSeasonRequests = seasonRequests.filter(
+        (seasonRequest) =>
+          (!seasonRequest.request.is4k &&
+            !seasonExists &&
+            (season.status === MediaStatus.AVAILABLE ||
+              season.status === MediaStatus.PARTIALLY_AVAILABLE)) ||
+          (seasonRequest.request.is4k &&
+            !seasonExists4k &&
+            (season.status4k === MediaStatus.AVAILABLE ||
+              season.status4k === MediaStatus.PARTIALLY_AVAILABLE))
+      );
+
+      let deletedSeason = false;
+
```

Comment:
Similar concept to our media requests checker. This will go through each season and make sure that season is available to prevent the wrong request deletion.